frontend/src/LineGraph.res

```diff
@@ -158,11 +161,33 @@ module Legend = {
     }
     switch xLabel {
     | Some(xLabel) =>
-      let html = "<b>" ++ (xLabel ++ "</b>")
-      let legend = Array.map((unit: t) => {
+      let html = "<b>" ++ xLabel ++ "</b>"
+      let row = (dashHTML, labelHTML, yHTML) => {
         "<div class='dygraph-legend-row'>" ++
-        (unit.dashHTML ++
-        ("<div>" ++ (" <b>" ++ (unit.yHTML ++ ("</b>" ++ ("</div>" ++ "</div>"))))))
+        (dashHTML ++
+        ("<div>" ++ labelHTML ++ "&ndash; </div>") ++
+        ("<div>" ++ " <b>" ++ yHTML ++ "</b>" ++ "</div>" ++ "</div>"))
+      }
+      let legend = Array.mapi((idx, unit: t) => {
+        // Add extra header for overall stats
+        let extraHeader = switch unit.label {
+        | "avg" => "<b>Overall Stats</b>"
+        | _ => ""
+        }
+        // We check if the data point was originally a multi-value data point,
+        // by testing if min and max are not Js.null. DataRow.single sets these
+        // to null, while they are set to non-null by DataRow.many
+        // When displaying Overall stats, we always have the min and max, which are
+        // displayed in the legend, even if the data-point is a single data point.
+        let (min, _, max) = Obj.magic(data.dygraph.rawData_[data.x])[idx + 1] // rawData_ also has x-index, so idx+1
```

Comment:
Maybe we don't need an overall min/max? I don't have strong opinions on this, but with multi-value data points it becomes quite confusing.

Reply:
I'm actually not sure we need the overall mean, either. Showing the overall stats in the legend is confusing to me. But, I think we can remove the overall min/max.

Reply:
![image](https://user-images.githubusercontent.com/315678/145563624-307559ac-0bcb-493a-9eb5-1cca2af79f96.png)

Screenshot without the min/max in the overall stats

Reply:
Yes, let's do that :)) thanks!